yellow-paper/docs/intro.md

```diff
@@ -24,6 +24,70 @@ The details should be sufficient for some other engineering team to implement th
 
 Some of the info we need to populate this document might have already been written in the top-level `docs/` dir of the monorepo. But the target audience is different. Reduce verbose prose. Remove monorepo code snippets (but note that simple pseudocode snippets to explain a protocol concept are fine). Don't describe components of the sandbox (that's an implementation detail and doesn't belong in this doc).
 
+## Diagrams
+To increase the probability of diagrams being up to date we encourage you to write them in `mermaid`. Mermaid is a markdown-like language for generating diagrams and is supported by Docusaurus, so it will be rendered automatically for you. 
+You simple create a codeblock specifying the language as `mermaid` and write your diagram in the codeblock. For example:
+```txt
+```mermaid
+graph LR
+    A --> B
+    B --> C
+    C --> A
+```
+
+```mermaid
+graph LR
+    A --> B
+    B --> C
+    C --> A
+```
+Mermaid support multiple types of diagrams so finding one that suite your needs should be possible. Consult their [documentation](https://mermaid.js.org/intro/getting-started.html) or try out their [live editor](https://mermaid.live/) to see if they got what you need.
+
+When writing class diagrams, we recommend using the `classDiagram` type and composition arrows `*--` to represent extensions. Also for the sake of readability, add all the components in the class itself, also if their are a composite. For example:
+
+```txt
```

Comment:
What does ```txt do?

Reply:
Just there fore rendering the txt verbosely such that the diagram was outlined in code and then rendered, in case people are looking at the doc and not looking at the source.